crates/rpc/src/rpc.rs

```diff
@@ -16,10 +16,23 @@ pub struct JsonRpcServer {
     cors_layer: Option<CorsLayer>,
     /// The [proxy layer](ProxyJsonRpcLayer) to forward requests.
     proxy_layer: Option<ProxyJsonRpcLayer>,
+    /// Whether to start an HTTP server.
+    http: bool,
+    /// Whether to start a WS server.
+    ws: bool,
     /// The RPC methods to be exposed.
     methods: Methods,
 }
 
+enum JsonRpcProtocolType {
+    /// Both HTTP and WS.
+    Both,
+    /// Only HTTP.
+    OnlyHttp,
```

Comment:
I think only `Http` is fine.

Reply:
thank you.
it is fixed [`27e5d39` (#195)](https://github.com/Vid201/silius/pull/195/commits/27e5d395bc338e5c0d9d685a7072c1c3d70aeaeb)